application/models/post.go

```diff
@@ -992,3 +1062,28 @@ func (p *Post) IsVisible(ctx context.Context, user User) bool {
 	}
 	return len(posts) > 0
 }
+
+func (p *Post) GetCurrentActions(user User) ([]string, error) {
+	transitions, err := p.GetStatusTransitions(user)
+	if err != nil {
+		return []string{}, err
+	}
+
+	allActions := statusActions()
+
+	actions := []string{}
+	for _, t := range transitions {
+		if action := allActions[t.Status]; action != "" {
+			actions = append(actions, action)
+		}
+	}
```

Comment:
Maybe move this loop into `statusActions` and then the return statement can be `return append(statusActions(transitions), providerActions...), nil` just to encapsulate functionality into smaller pieces.

Reply:
I think I'll leave it as is.  I don't see what this would gain us.  As I have it, `statusActions()` just returns a map. If I make this change, it would just move a tiny bit of processing from one function to another one that at this point doesn't include any.